### ISSUE TITLE
Flow rate Calibration test puts top layers on which does not show the flow anymore

### ISSUE THREAD

**Issue by Clankcoll** [SoftFever/OrcaSlicer#7422]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.2.0

### Operating System (OS)

Windows

### OS Version

10 | 22H2

### Printer

Sidewiner X2

### How to reproduce

Go to Calibration > Flow Rate > Pass 1 

### Actual results

The Model it loads are 10 Layers high and so laying the flowrate test over with an Covering layer.

Chaning Top shell layer to 0 does not resolve this problem since now the flow layer is missing completely

![image](https://github.com/user-attachments/assets/d20cb9d8-7b24-41f1-b470-20b69896269f)

![image](https://github.com/user-attachments/assets/5acbc425-b74c-449a-ae6a-2dca88326d26)

### Expected results

Have the flow layer as the top layer not covered in additional layers

### Project file & Debug log uploads

[Flowrate Test - Pass1.zip](https://github.com/user-attachments/files/17685057/Flowrate.Test.-.Pass1.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by igiannakas** [comment]
The shades of blue are different which means the flow is different. Try moving the horizontal slider on the top layer and you should see different flow volumes for each block.

**Comment by Auturistic** [comment]
Hello, I've noticed this today as well and made an account specifically to post this issue. Only some layers have the flow change.
I'm using Bambu P1S

I'm assuming the filament you are trying to configure uses generic PLA as a base for changes.

I've noticed that when I set the filament to Bambu PLA profile the contrast shows up as it should:
![image](https://github.com/user-attachments/assets/a6aa4128-3ce4-4c14-b3fc-d9ca25140687)

But when I switch to Generic PLA profile (the one I want to calibrate):
![image](https://github.com/user-attachments/assets/6445668f-93a2-4fbf-8e17-202b7f0ac53c)

So I dug a bit and found out that the only difference between these profiles is Max Volumetric Speed. Bambu has 21, while Generic is 12. When you switch Generic's MVS to 21 suddenly the contrast is back as it should. Btw this is not exclusive to Test1, YOLO and all others behave the same way.

I also noticed that Volumetric flow rate values on the side are wildly different (understandably):
21 mm3/s MVS:
![image](https://github.com/user-attachments/assets/78899ce9-cf1c-4f52-bad3-5ec401481e8d)
12 mm3/s MVS:
![image](https://github.com/user-attachments/assets/7cea6968-6457-47a3-b256-14f3d0215dc8)

My guess is that the slicer sets the values to the maximum it can given the set limit, but this calls in question the validity of this calibration because the difference between -20 and +20 should be striking. While as it stands now it tells the printer to print all the tests with the same flow.

In my case with Generic PLA profile, completely unchanged, it only prints layers 1 and 6 with different flow rate (same way your second picture shows). All other layers have the same flow rate (I checked the horizontal bar). Which makes this test pointless. Had I not checked this in the slicer I'd be wracking my head why all of them look the same.

Only at around 15 mm3/s MVS there is some change in other layers:
![image](https://github.com/user-attachments/assets/fe602996-f1cc-4c49-a586-7d82bc09d2c4)

**Comment by Auturistic** [comment]
Hi, so I think I figured it out. I think everything *is* working as intended. BUT the preview does not show it. Because the preview does not show the actual flow, only the theoretical one. I'll try to explain it but Tl;Dr: You can run your tests with no changes and it should work.

So when you look at my screenshots the one that is full red has a constant flow. That much is true. BUT I failed to notice that each slab is printed with different *Speed*. This makes sense, since the MVS value is low it can't exceed it because that value basically states that the filament can't melt efficiently with faster speeds than the one set. So to solve that the slicer is pretty smart and it varies the move speed instead.
**Speed Changes:**
![image](https://github.com/user-attachments/assets/05254e96-cf61-4296-896b-82f15f891a79)
![image](https://github.com/user-attachments/assets/3e924c55-ed94-4068-a441-3e231fda137e)

With the Bambu PLA profile the MVS value is high enough that the slicer decides that it can print at full speed and instead it will vary the flow from the extruder's side. So the speed stays at a constant while the flow value changes. Because slicer visualizes the flow value that's why in one there is a gradient of color and in the other there's not.
**Flow Changes:**
![image](https://github.com/user-attachments/assets/eb9f3e0d-4ba3-4b58-a3a0-62629a5e0245)
![image](https://github.com/user-attachments/assets/af34b7d9-d198-4c94-b7cd-0c0b1bf3d1d7)

You can check it by being on the top layer (with your regular setting, the ones that don't vary the color much) and pay attention to speed value at the horizontal bar at the bottom when you move from one slab to another. In my case it changes a lot as expected with -20 and +20 etc.

Also the reason why layers 1 and 6 show gradient is because they are of different type and they work a bit differently but it still varies the flow. 1 is bottom layer of course and 6 in the preview shows as a internal bridge because it a first layer right on the infill. Both of these types are printed at a much slower speed so the slicer can vary the flow and it shows up in the preview.

### **So in short this isn't a bug report, instead it should be a feature request for a better flow previews, kind of like the one prusa slicer has.**

**Comment by Auturistic** [comment]
I mean I can make some sure but I just use defaults. Every time I want to examine them I just make new ones by using the calibration menu. I just have everything at default and let the Test 1 change anything it wants. I just used Test 1 because it should exhibit higher contrast than the YOLO test that caused me to question things when everything was the same color. One more thing of note... I don't know why the print order is not defined to be rising or lowering flowrate. That would make sense to me for these test. I guess it doesn't matter that much since the temp stays the same.

[FlowTestProfiles.zip](https://github.com/user-attachments/files/17736355/FlowTestProfiles.zip)

1 is regular unmodified Bambu PLA Basic profile
2 is unmodified Generic PLA profile
3 is modified Generic PLA profile to have the same Max Volumetric Speed as the Bambu PLA Basic

**Comment by Clankcoll** [comment]
hmmm. 

But my prints of the test look nearly identical i cant really see a difference so that means i should get a proper mvs value and then try it again i guess.

**Comment by Auturistic** [comment]
Hi, I have been wrong with my assumptions. At this point I have no clue how OrcaSlicer decides these things anymore... Someone with actual inside knowledge on how the slicer works would have to chime in. 

Try this test instead: https://makerworld.com/en/models/189543

It correctly shows flow changes in the slicer. I have yet to compare the results to the built in tests but this is bizarre. I expected it to behave the same way. The linked test shows the correct flow even when you change the top pattern so that's not it.

To make me lose my mind more; I set up the Pass1 test after this the same way I always do to compare settings... and it shows the correct flow gradient with constant speed. Even when I change the layer height profiles but keep the changes the flow test forces it works correctly now. Yet when I open the files I uploaded here it is still the same it was - it shows the change in speed but constant flow.

I don't understand at all. It's the same Generic PLA profile, the exact same, I combed the settings one by one. I went through all the settings and I figured out why it suddenly works for me. I *believe* this is a different issue to the one you have so I will make a separate bug report with more detailed explanation. If you are interested: https://github.com/SoftFever/OrcaSlicer/issues/7509

So yeah, if this relates to your problem in any way... try opening up a new instance of Orca every time you want to test/ change some settings... what a bizarre issue.

**Comment by HEPBO3AH** [comment]
I spent several hours last night dealing with this. I can confirm that everything you found was correct. 

The issue for me only happens on Pass 2 and it only happens if Pass 1 shows improvement in the range +10 to +20.

It would be good if the team could chime in.
